lib/magic_pipe/transports/https.rb

```diff
@@ -45,8 +46,13 @@ def url
         @options.fetch(:url)
       end
 
-      def basic_auth_user
-        @options.fetch(:basic_auth_user)
+      def basic_auth_user(topic)
+        user_auth = @options.fetch(:basic_auth_user)
+        if user_auth.respond_to?(:call)
+          user_auth.call(topic)
+        else
+          user_auth
+        end
```

Comment:
Ok, but then we should do the same for the password as well. It might be easier to just combine them in a single config value `http_basic_auth` that accepts a `user:password` string. The callable would be expected to return the same kind of string.